Tests/NIOTests/LengthFieldBasedFrameDecoderTest.swift

```diff
@@ -0,0 +1,91 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+//
+import XCTest
+
+struct fixedLengthFrameDecoderHelper<T: FixedWidthInteger> {
```

Comment:
- the type name should be upper case
- this helper type should go to the end of the file. In Xcode it's really helpful is the test class itself is at the top as it contains the 'run' icon for the whole test suite right next to the class. If it's not at the top it's always kind of hard to find the test class as you need to scroll and look for it...